### PR TITLE
DUPLO-26546, DUPLO-26566: New attribute support for azure node pool K8sWorkerOs, NodeTaints, NodeLabels.

### PR DESCRIPTION
**Pull request by tahir-duplo**
## Overview

- New attribute support for azure node pool K8sWorkerOs, NodeTaints, NodeLabels.

## Summary of changes

This PR does the following:

- New attribute support for azure node pool K8sWorkerOs, NodeTaints, NodeLabels.
- Two new resources are added, `duplocloud_azure_tenant_key_vault` and `duplocloud_azure_tenant_key_vault_secret`

## Testing performed

- [ ] Using unit tests
- [x] Manually, on my local system
- [ ] Manually, on a remote test system

## Describe any breaking changes

- Backward compatibility of azure nodepool.

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-26566 Add support of new resource duplocloud_azure_tenant_key_vault for pathlock](https://app.clickup.com/t/8655600/DUPLO-26566)

File: duplocloud/provider.go [hunk 1]
```diff
@@ -120,6 +120,8 @@ func Provider() *schema.Provider {
 			"duplocloud_aws_sns_topic":                      resourceAwsSnsTopic(),
 			"duplocloud_aws_lb_listener_rule":               resourceAwsLbListenerRule(),
 			"duplocloud_azure_key_vault_secret":             resourceAzureKeyVaultSecret(),
+			"duplocloud_azure_tenant_key_vault":             resourceAzureTenantKeyVault(),
+			"duplocloud_azure_tenant_key_vault_secret":      resourceAzureTenantKeyVaultSecret(),
```
Comment (Duplo-nikv) [inline duplocloud/provider.go:124]: this came from which PR

**Review by Duplo-nikv — COMMENTED** [review]
There is a ticket DUPLO-21437 related to key vaults which is not started

File: duplocloud/provider.go (hunk 1 above)
Comment (tahir-duplo) [reply duplocloud/provider.go:124]: @Duplo-nikv Same PR for DUPLO-26546, DUPLO-26566

File: duplocloud/resource_duplo_azure_tenant_key_vault.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"enabled_for_disk_encryption": {
+			Description: "Azure Disk Encryption is permitted to retrieve secrets from the vault and unwrap keys.",
+			Type:        schema.TypeBool,
+			Computed:    true,
+		},
+	}
+}
```
Comment (Duplo-nikv) [inline duplocloud/resource_duplo_azure_tenant_key_vault.go:68]: If there is no update context can every attribute be force new. Or suppress change , whichever best fit

**Review by Duplo-nikv — CHANGES_REQUESTED** [review]

**Review by Duplo-nikv — APPROVED** [review]
LGTM

**Pull request merged**